src/request/RequestClient.ts

```diff
@@ -0,0 +1,444 @@
+import axios, { AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios'
+import { CsrfToken, JobExecutionError } from '..'
+import { isAuthorizeFormRequired, isLogInRequired } from '../auth'
+import { LoginRequiredError } from '../types'
+import { AuthorizeError } from '../types/AuthorizeError'
+import { NotFoundError } from '../types/NotFoundError'
+import { parseWeboutResponse } from '../utils/parseWeboutResponse'
+
+export interface HttpClient {
+  get<T>(
+    url: string,
+    accessToken: string | undefined,
+    contentType: string,
+    overrideHeaders: { [key: string]: string | number }
+  ): Promise<{ result: T; etag: string }>
+
+  post<T>(
+    url: string,
+    data: any,
+    accessToken: string | undefined,
+    contentType: string,
+    overrideHeaders: { [key: string]: string | number }
+  ): Promise<{ result: T; etag: string }>
+
+  put<T>(
+    url: string,
+    data: any,
+    accessToken: string | undefined,
+    overrideHeaders: { [key: string]: string | number }
+  ): Promise<{ result: T; etag: string }>
+
+  delete<T>(
+    url: string,
+    accessToken: string | undefined
+  ): Promise<{ result: T; etag: string }>
+
+  getCsrfToken(type: 'general' | 'file'): CsrfToken | undefined
+  clearCsrfTokens(): void
+}
+
+export class RequestClient implements HttpClient {
+  private csrfToken: CsrfToken = { headerName: '', value: '' }
+  private fileUploadCsrfToken: CsrfToken | undefined
+  private httpClient: AxiosInstance
+
+  constructor(private baseUrl: string, allowInsecure = false) {
+    const https = require('https')
+    this.httpClient = axios.create({
+      baseURL: baseUrl,
+      httpsAgent: new https.Agent({
```

Comment:
![image](https://user-images.githubusercontent.com/18329105/107035375-6325d800-67b8-11eb-8241-64ca753afda1.png)

The error in angular (Data Controller)

Reply:
This was happening because `Agent` can only be instantiated in Node and not in the browser. I've added a check to the code to prevent the error.